toponetx/classes/combinatorial_complex.py

```diff
@@ -528,3 +776,412 @@ def clone(self) -> "CombinatorialComplex":
         for cell in self.cells:
             CC.add_cell(cell, self.cells.get_rank(cell))
         return CC
+
+    def is_connected(self, s=1, cells=False):
+        """Determine if combintorial complex is :term:`s-connected <s-connected, s-node-connected>`.
+
+        Parameters
+        ----------
+        s : int, list, optional, default : 1
+            Minimum number of edges shared by neighbors with node.
+
+        cells: boolean, optional, default: False
+            If True, will determine if s-cell-connected.
+            For s=1 s-cell-connected is the same as s-connected.
+
+        Returns
+        -------
+        is_connected : boolean
+
+        Notes
+        -----
+        A CHG is s node connected if for any two nodes v0,vn
+        there exists a sequence of nodes v0,v1,v2,...,v(n-1),vn
+        such that every consecutive pair of nodes v(i),v(i+1)
+        share at least s cell.
+
+        Examples
+        --------
+        >>> CHG = ColoredHyperGraph(cells=E)
+        >>> CHG.is_connected()
+        """
+        B = self.incidence_matrix(rank=0, to_rank=None, incidence_type="up")
+        if cells:
+            A = incidence_to_adjacency(B, s=s)
+        else:
+            A = incidence_to_adjacency(B.transpose(), s=s)
+        G = nx.from_scipy_sparse_matrix(A)
+        return nx.is_connected(G)
+
+    def singletons(self):
+        """Return a list of singleton cell.
+
+        A singleton cell is a cell of
+        size 1 with a node of degree 1.
+
+        Returns
+        -------
+        singles : list
+            A list of cells uids.
+        """
+        singletons = []
+        for cell in self.cells:
+            zero_elements = self.cells[cell].skeleton(0)
+            if len(zero_elements) == 1:
+                for n in zero_elements:
+                    if self.degree(n) == 1:
+                        singletons.append(cell)
+        return singletons
```

Comment:
I think a generator would be a better choice here. If the caller wants a list, they can always just construct one.

```suggestion
        for cell in self.cells:
            zero_elements = self.cells[cell].skeleton(0)
            if len(zero_elements) == 1:
                for n in zero_elements:
                    if self.degree(n) == 1:
                        yield cell
```